src/python/pants/option/global_options.py

```diff
@@ -572,7 +572,8 @@ def register_bootstrap_options(cls, register):
             "--print-exception-stacktrace",
             advanced=True,
             type=bool,
-            fingerprint=False,
+            default=False,
+            daemon=True,
             help="Print to console the full exception stack trace if encountered.",
```

Comment:
See #10035 for the explanation of when `fingerprint=True` vs `daemon=True` is relevant, but: afaict exceptions before the scheduler is created should always result in a full stacktrace, so it shouldn't be necessary to restart `pantsd` (ie, `daemon=True`) for this option.

On the other hand, having this marked `fingerprint=False` means that we won't recreate the `Scheduler` if this value changes... and _that_ is a bug, because the `Scheduler` does consume this option. So I think that you probably just want to mark this `fingerprint=True`.

Reply:
Sounds good. This morning I was having issues with `fingerprint=True` actually causing a change in behavior - it did change the behavior, but it looked like portions of the stacktrace were still being printed (less than normal). I didn't capture that output, sorry.

I can't reproduce it though, so looks like this is working correctly now.

Reply:
> I can't reproduce it though

That would be because I changed the wrong option. Oops.

This is the output I get when `fingerprint=True, daemon=False`. 

```
17:18:41 [ERROR] 1 Exception encountered:

  ValueError:
Traceback (most recent call last):
  File "/Users/eric/DocsLocal/code/projects/pants/src/python/pants/bin/local_pants_runner.py", line 288, in run
    engine_result = self._run_v2()
  File "/Users/eric/DocsLocal/code/projects/pants/src/python/pants/bin/local_pants_runner.py", line 176, in _run_v2
    return self._maybe_run_v2_body(goals, poll=False)
  File "/Users/eric/DocsLocal/code/projects/pants/src/python/pants/bin/local_pants_runner.py", line 199, in _maybe_run_v2_body
    poll_delay=(0.1 if poll else None),
  File "/Users/eric/DocsLocal/code/projects/pants/src/python/pants/init/engine_initializer.py", line 135, in run_goal_rules
    goal_product, params, poll=poll, poll_delay=poll_delay
  File "/Users/eric/DocsLocal/code/projects/pants/src/python/pants/engine/internals/scheduler.py", line 558, in run_goal_rule
    self._raise_on_error([t for _, t in throws])
  File "/Users/eric/DocsLocal/code/projects/pants/src/python/pants/engine/internals/scheduler.py", line 528, in _raise_on_error
    wrapped_exceptions=tuple(t.exc for t in throws),
pants.engine.internals.scheduler.ExecutionError: 1 Exception encountered:

  ValueError:
```

If I have `daemon=True`, I get the expected result:

```
▶ ./pants --no-pantsd --no-print-exception-stacktrace lint src/python/pants/util/strutil.py
17:18:34 [ERROR] 1 Exception encountered:

  ValueError:

(Use --print-exception-stacktrace to see more error details.)
```

Reply:
Hm. It looks like that's because this is a sneaky global-mutable option on the `ExceptionSink`: https://github.com/pantsbuild/pants/blob/9ca7b52b49d4e64c59d02278659ce36a4c9a787f/src/python/pants/init/logging.py#L70-L76 ... (which I added. *sigh) would you mind adding a comment/TODO explaining that and setting `daemon=True` for now?